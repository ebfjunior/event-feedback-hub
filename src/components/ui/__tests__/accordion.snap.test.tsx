import React from 'react';
import { render } from '@testing-library/react';
import { Accordion, AccordionContent, AccordionItem, AccordionTrigger } from '../accordion';

vi.mock('lucide-react', async () => {
	const actual = await vi.importActual<any>('lucide-react');
	return { ...actual, ChevronDownIcon: (p: any) => <svg data-icon="chev-down" {...p} /> };
});

describe('ui/Accordion (snapshot)', () => {
	it('renders basic accordion', () => {
		const { container } = render(
			<Accordion type="single" collapsible>
				<AccordionItem value="item-1">
					<AccordionTrigger>Trigger</AccordionTrigger>
					<AccordionContent>Content</AccordionContent>
				</AccordionItem>
			</Accordion>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});


