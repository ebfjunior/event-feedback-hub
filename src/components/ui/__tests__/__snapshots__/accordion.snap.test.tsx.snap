// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/Accordion (snapshot) > renders basic accordion 1`] = `
<div
  data-orientation="vertical"
  data-slot="accordion"
>
  <div
    class="border-b last:border-b-0"
    data-orientation="vertical"
    data-slot="accordion-item"
    data-state="closed"
  >
    <h3
      class="flex"
      data-orientation="vertical"
      data-state="closed"
    >
      <button
        aria-controls="radix-«r1»"
        aria-expanded="false"
        class="focus-visible:border-ring focus-visible:ring-ring/50 flex flex-1 items-start justify-between gap-4 rounded-md py-4 text-left text-sm font-medium transition-all outline-none hover:underline focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&[data-state=open]>svg]:rotate-180"
        data-orientation="vertical"
        data-radix-collection-item=""
        data-slot="accordion-trigger"
        data-state="closed"
        id="radix-«r0»"
        type="button"
      >
        Trigger
        <svg
          class="text-muted-foreground pointer-events-none size-4 shrink-0 translate-y-0.5 transition-transform duration-200"
          data-icon="chev-down"
        />
      </button>
    </h3>
    <div
      aria-labelledby="radix-«r0»"
      class="data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down overflow-hidden text-sm"
      data-orientation="vertical"
      data-slot="accordion-content"
      data-state="closed"
      hidden=""
      id="radix-«r1»"
      role="region"
      style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
    />
  </div>
</div>
`;
